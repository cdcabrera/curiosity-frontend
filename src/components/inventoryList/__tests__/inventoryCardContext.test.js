import { context, useGetInstancesInventory } from '../inventoryCardContext';

describe('InventoryCardContext', () => {
  it('should return specific properties', () => {
    expect(context).toMatchSnapshot('specific properties');
  });

  it('should handle instances inventory API responses', () => {
    const { result: errorResponse } = shallowHook(() =>
      useGetInstancesInventory({
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useProductInventoryHostsQuery: () => ({}),
        useSelectorsResponse: () => ({ error: true })
      })
    );

    expect(errorResponse).toMatchSnapshot('instances inventory, error');

    const { result: pendingResponse } = shallowHook(() =>
      useGetInstancesInventory({
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useProductInventoryHostsQuery: () => ({}),
        useSelectorsResponse: () => ({ pending: true })
      })
    );

    expect(pendingResponse).toMatchSnapshot('instances inventory, pending');

    const { result: cancelledResponse } = shallowHook(() =>
      useGetInstancesInventory({
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useProductInventoryHostsQuery: () => ({}),
        useSelectorsResponse: () => ({ cancelled: true })
      })
    );

    expect(cancelledResponse).toMatchSnapshot('instances inventory, cancelled');

    const { result: fulfilledResponse } = shallowHook(() =>
      useGetInstancesInventory({
        useProduct: () => ({ productId: 'lorem' }),
        useDispatch: () => {},
        useProductInventoryHostsQuery: () => ({}),
        useSelectorsResponse: () => ({ fulfilled: true })
      })
    );

    expect(fulfilledResponse).toMatchSnapshot('instances inventory, fulfilled');
  });
});
