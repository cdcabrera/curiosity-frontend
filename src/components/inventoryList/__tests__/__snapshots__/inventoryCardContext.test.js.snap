// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryCardContext should handle instances inventory API responses: instances inventory, cancelled 1`] = `
Object {
  "data": Object {},
  "error": undefined,
  "fulfilled": undefined,
  "pending": true,
}
`;

exports[`InventoryCardContext should handle instances inventory API responses: instances inventory, error 1`] = `
Object {
  "data": Object {},
  "error": true,
  "fulfilled": undefined,
  "pending": false,
}
`;

exports[`InventoryCardContext should handle instances inventory API responses: instances inventory, fulfilled 1`] = `
Object {
  "data": Object {},
  "error": undefined,
  "fulfilled": true,
  "pending": false,
}
`;

exports[`InventoryCardContext should handle instances inventory API responses: instances inventory, pending 1`] = `
Object {
  "data": Object {},
  "error": undefined,
  "fulfilled": undefined,
  "pending": true,
}
`;

exports[`InventoryCardContext should return specific properties: specific properties 1`] = `
Object {
  "useGetInstancesInventory": [Function],
  "useOnColumnSortInstances": [Function],
  "useOnPageInstances": [Function],
}
`;
